lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,183 @@
+            logger.info("*** Message from sysadmins of %s:", results_webserver)
+            logger.info("***\n*** %s", message)
+            logger.info("***\n*** No local actions taken.")
+            sys.exit(1)
+        results_path_prefix = response.text.split(":")[1]
+        if not results_path_prefix:
+            logger.error(
+                "fetch results host info did not contain a path prefix: %s",
+                response.text,
+            )
+            sys.exit(1)
 
-    try:
         temp_dir = tempfile.mkdtemp(
-            dir=config.pbench_tmp, prefix="pbench-move-results."
+            dir=self.config.pbench_tmp, prefix="pbench-move-results."
         )
-    except Exception:
-        logger.error("Failed to create temporary directory")
-        sys.exit(1)
-
-    runs_copied = 0
-    failures = 0
-
-    for dirent in config.pbench_run.iterdir():
-        if not dirent.is_dir():
-            continue
-        if dirent.name.startswith("tools-") or dirent.name == "tmp":
-            continue
-        result_dir = dirent
-        mrt = MakeResultTb(result_dir, temp_dir, _user, _prefix, config, logger)
-        result_tb_name = mrt.make_result_tb()
-        assert (
-            result_tb_name
-        ), "Logic bomb!  make_result_tb() always returns a tar ball name"
-        crt = CopyResultTb(controller, result_tb_name, config, logger)
-        crt.copy_result_tb()
-        try:
-            # We always remove the constructed tar ball, regardless of success
-            # or failure, since we keep the result directory below on failure.
-            os.remove(result_tb_name)
-            os.remove(f"{result_tb_name}.md5")
-        except OSError:
-            logger.error("rm failed to remove %s and its .md5 file", result_tb_name)
-            sys.exit(1)
 
-        try:
-            shutil.rmtree(result_dir)
-        except OSError:
-            logger.error("rm failed to remove the %s directory hierarchy", result_dir)
-            sys.exit(1)
+        runs_copied = 0
+        failures = 0
 
-        runs_copied += 1
+        for dirent in config.pbench_run.iterdir():
+            if not dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+            result_dir = dirent
+            mrt = MakeResultTb(
+                result_dir,
+                temp_dir,
+                self.context.user,
+                self.context.prefix,
+                self.config,
+                logger,
+            )
 
-    if runs_copied + failures > 0:
-        logger.debug(
-            "successfully moved %s runs, encountered %s failures", runs_copied, failures
-        )
+            try:
+                make_return_val = mrt.check_result_target_dir(result_dir, temp_dir)
+            except FileNotFoundError as e:
+                logger.error(e)
+
+            if make_return_val:
+                failures += 1
+                continue
```

Comment:
As this code stands at the moment, if `FileNotFoundError` is raised and caught, then the code might not notice whether a failure occurred (depending on what the value of `make_return_val` is from a previous iteration).

If you convert `check_result_target_dir` to report _all_ errors via exceptions, then you can get rid of the conditional at line 81 and move the statements from the "then clause" into the `except` block at line 78; you may want to reconsider which exception you catch there (it might be worth setting up a class hierarchy of our own exceptions, as Dave suggested earlier, so that you can catch them all using a base class, in cases where you are less concerned about which one you got).

Reply:
Done